### ISSUE TITLE
Which Firefox version has the best compatibility with RAS? 

### ISSUE THREAD

**Issue by thesilent33** [dillbyrne/random-agent-spoofer#576]
So, RAS won't work with the latest Firefox release. So, which version of Firefox has the best compatibility with RAS? Thanks for your answer. :)

**Comment by dillbyrne** [comment]
I would recommend using the tor browser and or the brave browser both of which have anti fingerprinting and other privacy options built into them. RAS is old now and you would be at risk using an out of date version of firefox. The ESR version will stay compatible with RAS for a while longer until the next ESR release.

**Comment by thesilent33** [comment]
Yes, so sad that this add-on discontinued. Thanks for the advice. Btw, is there any alternative addon for RAS that can change the user agent randomly at every browser restart like RAS? I just need that particular feature. Thanks.

**Comment by B00ze64** [comment]
@dillbyrne Hey mate, so long and thanks for all the fish, as Douglas Adams says. RAS is a great tool and I know you put a LOT of work into it, so Thanks! Best Regards,

**Comment by neroux** [comment]
It should still work fine with the latest version for more than a month until v57 is out. Only then Mozilla will switch to web extensions only. You could also switch to 52 ESR which will support it until next June.

As for web extension alternatives, you could try https://github.com/tarampampam/random-user-agent though it is primarily Chrome-oriented and I believe there are issues with Firefox. Furthermore the original developer is not very active.

**Comment by gitbugged** [comment]
@Jhony33 It works well on [FF52 ESR](https://www.mozilla.org/en-US/firefox/organizations/). 👍 

I made an unofficial update based on @forkoz's fork with lots of UA updates. You can load it like so...

To load the unsigned addon in FF52:
   -  Load about:config in Firefox's address bar and hit enter.
   -  Confirm you will be careful if a warning message is displayed.
   -  Search for xpinstall.signatures.required.
   -  Double-click on the preference name so that its value is set to false.
   - Restart browser
   - Go to Tools -> Addons
   - Download (10/11/17-unofficial): [random_agent_spoofer-0.9.5.7-fx.zip](https://github.com/dillbyrne/random-agent-spoofer/files/1376895/random_agent_spoofer-0.9.5.7-fx.zip), and rename to .xpi instead of .zip
  - Drag + Drop the XPI onto the Addons page. Allow the installation.

- Also supports Pale Moon.

**Comment by LimboSlam** [comment]
@Jhony33, do you mean it is compatible with Pale Moon or added specific, Pale Moon, user agent strings to the add-on settings? 

If proper support is ever implemented, please review following: https://developer.palemoon.org/Add-ons:Extensions/Jetpack (PMkit documentation draft). I wouldn't be discourage, because "several popular add-ons have already been [successfully adapted](https://github.com/JustOff/pm27-sdk-addons/blob/master/README.md#sdk-based-add-ons-can-be-run-in-pale-moon-271-but-they-should-be-properly-adapted) and work fine in Pale Moon."

**Comment by Jhony33** [comment]
@gitbugged Wow, that's cool man! Will it work with the newest Firefox version too?

**Comment by thesilent33** [comment]
@gitbugged Wow, that's cool man! Will it work with the newest Firefox version too?
@LimboSlam No, I meant the previous version (old version) of Firefox.

**Comment by neroux** [comment]
@Jhony33 No, it will only work up to 56. Unless you want to stay on that version your best bet is 52 ESR for the time being.

**Comment by thesilent33** [comment]
@neroux Okay, thanks for the answer. I'll consider using 52 ESR.

**Comment by neroux** [comment]
Just keep in mind, 52 ESR will be out of support at the end of next June too. So if security updates are important to you, you might have to migrate at some point in July or August.

**Comment by gotowf** [comment]
IMHO the best Firefox version for running legacy extensions is [Waterfox](https://www.waterfoxproject.org/), which has no plan of dropping legacy extensions and is currently building an independent add-on store to save them.

**Comment by gitbugged** [comment]
New Version released for FF52 ESR browsers. 
https://github.com/gitbugged/random-agent-spoofer/releases/tag/0.9.5.8.1

**Comment by gitbugged** [comment]
@neroux Per Pale Moon team they will continue providing security updates to FF52 ESR on their UXP branch. This should buy us even more time. I doubt RAS will ever be ported to FF60 ESR due to massive incompatibilities in the API. I would tend to say FF52 is safer for the time being.

**Comment by neroux** [comment]
Thanks for the heads up @gitbugged. In the meantime I have also made the extension mentioned earlier available for Firefox
https://addons.mozilla.org/firefox/addon/random_user_agent/

**Comment by FrostKnight** [comment]
https://wiki.hyperbola.info/doku.php?id=en:project:iceweasel-uxp_addons

if you want an updated version that works with basilisk uxp/palemoon uxp
